test/es-module/test-esm-transform-source-loader.mjs

```diff
@@ -0,0 +1,6 @@
+// Flags: --experimental-loader ./test/fixtures/es-module-loaders/transform-source.mjs
+/* eslint-disable node-core/require-common-first, node-core/required-modules */
+import assert from 'assert';
+import { message } from '../fixtures/es-modules/message.mjs';
+
+assert.strictEqual(message, 'A MESSAGE');
```

Comment:
this should have an example where the transforms returns a format and it differs from the argument passed in